apps/judicial-system/web/src/utils/hooks/useS3Upload/useS3Upload.ts

```diff
@@ -294,14 +293,18 @@ const useS3Upload = (caseId: string) => {
             newFileId,
           )
 
-          filesUploaded.push(file)
+          return true
         } catch (e) {
           toast.error(formatMessage(strings.uploadFailed))
-          updateFile({ ...file, status: 'error' })
+          updateFile({ ...file, percent: 0, status: 'error' })
+
+          return false
         }
-      }
+      })
 
-      return filesUploaded.length === files.length
+      return Promise.all(promises).then((results) =>
+        results.every((result) => result),
+      )
```

Comment:
For components that need to know the overall upload success, this aggregates the results of all uploads.